src/map/mob.cpp

```diff
@@ -2217,6 +2269,13 @@ static int32 mob_ai_sub_lazy(struct mob_data *md, va_list args)
 	if (DIFF_TICK(tick, md->last_skillcheck) < MOB_SKILL_INTERVAL)
 		return 0;
 
+	// Check for warp chase if there's still a target
```

Comment:
Did I miss something above, or why are we assuming that the target is not on the same map (or not even in range) here anymore?

Reply:
It does so pretty late and with additional distance checking.
Do we really want it this way?

Reply:
I sorted the checks in mob_warpchase so that they are in the optimal order (at least as to the best of my knowledge).

The distance checking happens at the last moment and only if the map is equal and all other requirements for warp chase are met.

None of these checks are very cost-intenstive and we only call mob_warpchase once and then unlock target, so as long as we map sure not to reach map_foreachinallrange unless we already have checked everything else, we should be fine.